packages/frontend/src/components/ControlledAutocomplete/index.tsx

```diff
@@ -76,55 +80,92 @@ function ControlledAutocomplete(
-                  )
-                }
-              >
-                {label}
-              </FormLabel>
-            )}
-            {/* Dropdown row option content */}
-            <Flex>
-              <Box flexGrow={1}>
-                <SingleSelect
-                  name="choose-dropdown-option"
-                  colorScheme="secondary"
-                  isClearable={!required}
-                  items={items}
-                  onChange={onChange}
-                  value={fieldValue}
-                  placeholder={placeholder}
-                  ref={ref}
-                  data-test={`${name}-autocomplete`}
-                  onRefresh={onRefresh}
-                  isRefreshLoading={loading}
-                  freeSolo={freeSolo}
-                />
-              </Box>
-            </Flex>
-            {isError && (
-              <FormErrorMessage>{fieldState.error?.message}</FormErrorMessage>
-            )}
-          </FormControl>
-        )
-      }}
-    />
+  // Control add new option modal
+  const {
+    onClose: onNewOptionModalClose,
+    onOpen: onNewOptionModalOpen,
+    isOpen: isNewOptionModalOpen,
+  } = useDisclosure()
+
+  const { createNewOption, isCreatingNewOption } =
+    useCreateNewOption(fieldOnChange)
+
+  const onNewOptionModalSubmit = useCallback(
+    (inputValue: string) => {
+      onNewOptionModalClose()
+      createNewOption({ inputValue, addNewId: addNewOption?.id })
+    },
+    [addNewOption?.id, createNewOption, onNewOptionModalClose],
+  )
+
+  return (
+    <FormControl isInvalid={isError}>
+      {label && (
+        <FormLabel
+          isRequired={required}
+          description={
+            description && (
+              <Markdown linkTarget="_blank">{description}</Markdown>
+            )
+          }
+        >
+          {label}
+        </FormLabel>
+      )}
+      {/* Dropdown row option content */}
+      <Flex>
+        <Box flexGrow={1}>
+          <SingleSelect
+            name="choose-dropdown-option"
+            colorScheme="secondary"
+            isClearable={!required}
+            items={items}
+            onChange={fieldOnChange}
+            value={fieldValue}
+            placeholder={placeholder}
+            ref={ref}
+            data-test={`${name}-autocomplete`}
+            onRefresh={onRefresh}
+            isRefreshLoading={loading}
+            freeSolo={freeSolo}
+            addNew={
+              addNewOption
+                ? {
+                    type: addNewOption.type,
+                    label: addNewOption.label,
+                    onSelected: onNewOptionModalOpen,
+                    isCreating: isCreatingNewOption,
+                  }
+                : undefined
+            }
+          />
+        </Box>
+      </Flex>
+      {isError && <FormErrorMessage>{error?.message}</FormErrorMessage>}
+      {addNewOption?.type === 'modal' && isNewOptionModalOpen && (
+        <AddNewOptionModal
```

Comment:
nit: probably can add comment to say we remount (instead of using the `isOpen` prop) because less dev effort to clear form data that way <.<. But not as performant

Reply:
true